### PR TITLE
Add RPC clients tests

### PR DESCRIPTION
**Pull request by wirednkod**

### CODE REVIEW

File: rpc/providers/proxy.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { KnownRpcMethods } from "../../known/mod.ts";
+import { proxyClient, ProxyBeacon } from "./proxy.ts"
+import { node } from "../../test-util/node.ts";
+import { assert, assertEquals } from "../../_deps/asserts.ts";
+
+Deno.test({
+  name: "Test proxClient flow",
```
Comment (harrysolovay) [inline rpc/providers/proxy.test.ts:7]: typo

File: rpc/providers/proxy.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        const raw = await client.call("state_getMetadata", [])
+        assert(raw.result)
+      }
+    })
+
+    // even though CAPI is not allowing this so this tests should probably not be here
+    await s.step({
+      name: "Tests error when a call is made with wrong method",
+      fn: async () => {
+        const raw = await client.call("state_getMetaSomething" as keyof KnownRpcMethods, [])
```
Comment (harrysolovay) [inline rpc/providers/proxy.test.ts:34]: Trigger the type error and annotate with [`// @ts-expect-error`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments)

**Comment by wirednkod** [comment]
I just finalized the tests for RPC and was trying to figure out how to run them but realized that since our last conversation you proceed and wrote the tests yourself. I am not sure why but that makes this PR useless.

It would be nice to let me know next time, so that noone is stepping in other people's task

**Comment by harrysolovay** [comment]
Closing per conversation with @wirednkod, who lacks the bandwidth to push this to completion.

**Pull request closed without merging**